depthai_ros_driver/include/depthai_ros_driver/types.hpp

```diff
@@ -0,0 +1,107 @@
+#pragma once
+
+#include <stdexcept>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+namespace depthai_ros_driver {
+namespace types {
+
+enum class NNType{
+    None,
+    Default,
+    Spatial
+};
+namespace cam_types {
+enum class CamName { OAK_D, OAK_D_W, OAK_D_PRO_AF, OAK_D_PRO_W, OAK_D_S2, OAK_D_LITE, OAK_1, OAK_1_W, OAK_1_LITE, OAK_1_MAX };
+std::unordered_map<std::string, CamName> cam_names = {
+    {"OAK-D", CamName::OAK_D},
+    {"OAK-D-W", CamName::OAK_D_W},
+    {"OAK-D-PRO-AF", CamName::OAK_D_PRO_AF},
+    {"OAK-D-PRO-W", CamName::OAK_D_PRO_W},
+    {"OAK-D-S2", CamName::OAK_D_S2},
+    {"OAK-D-LITE", CamName::OAK_D_LITE},
+    {"OAK-1", CamName::OAK_1},
+    {"OAK-1-W", CamName::OAK_1_W},
+    {"OAK-1-LITE", CamName::OAK_1_LITE},
+    {"OAK-1-MAX", CamName::OAK_1_MAX},
+};
+class CamType {
+   public:
+    explicit CamType(const std::string& name) {
+        auto cam_name_enum = cam_names.at(name);
+        switch(cam_name_enum) {
+            case CamName::OAK_D:
+                stereo_ = true;
+                imu_available_ = true;
+                ir_available_ = false;
+                break;
+            case CamName::OAK_D_W:
+                stereo_ = true;
+                imu_available_ = true;
+                ir_available_ = false;
+                break;
+            case CamName::OAK_D_PRO_AF:
+                stereo_ = true;
+                imu_available_ = true;
+                ir_available_ = true;
+                break;
+            case CamName::OAK_D_PRO_W:
+                stereo_ = true;
+                imu_available_ = true;
+                ir_available_ = true;
+                break;
+            case CamName::OAK_D_S2:
+                stereo_ = true;
+                imu_available_ = true;
+                ir_available_ = false;
+                break;
+            case CamName::OAK_D_LITE:
+                stereo_ = true;
+                imu_available_ = true;
+                ir_available_ = false;
+                break;
+            case CamName::OAK_1:
+                stereo_ = false;
+                imu_available_ = false;
+                ir_available_ = false;
+                break;
+            case CamName::OAK_1_W:
+                stereo_ = false;
+                imu_available_ = false;
+                ir_available_ = false;
+                break;
+            case CamName::OAK_1_LITE:
+                stereo_ = false;
+                imu_available_ = false;
+                ir_available_ = false;
+                break;
+            case CamName::OAK_1_MAX:
+                stereo_ = false;
+                imu_available_ = false;
+                ir_available_ = false;
```

Comment:
These things should be discoverable by querying a device.
If they are not, core should expose them.

(getIrDrivers, getConnectedCameraFeatures & afaik imu is missing indeed. CC: @szabi-luxonis )